benchmarking/smartseq2/picard_qc_analysis/QC_metrics_analysis.R

```diff
@@ -0,0 +1,154 @@
+# R script to do metrics test 
+# loading library
+library(ggplot2)
+library(MASS)
+library(plyr)
+library(reshape2)
+library(ggpubr)
+library(gridExtra)
+library(ggpmisc)
+library(cowplot)
+
+##convert p-value to star char
+convert2star<-function(x){
+  if(x>0.05){
+    p='ns'
+  }else if(x> 0.01){
+    p='*'
+  }else if(x>0.001){
+    p="**"
+  }else if(x>0.0001){
+    p='***'
+  }else{
+    p='****'
+  }
+  return(p)
+}
+## standard theme
+addTheme<-function(p){
+  p<-p+theme(legend.position ="top")
+  p<-p+theme(plot.title = element_text(hjust = 0.5,size=20,face='bold'))
+  p<-p+theme(axis.title.x = element_text(color='black',size=18,face='bold'),axis.title.y = element_text(size=18,color='black',face='bold'))
+  p<-p+theme(axis.text.y = element_text(color='black',size=18,face='bold'),axis.text.x = element_text(color='black',size=18,face='bold'))
+  p<-p+theme(legend.text=element_text(size=15,face='bold'),legend.title = element_blank())
+  return(p)
+}
+
+# metrics files
+# metfile1 is the production pipeline results
+# metfile2 is the updated pipeline results
+args <-commandArgs(trailingOnly=TRUE)
+print(args)
+metfile1<-args[1]
+metfile2<-args[2]
+#  output prefix
+output_name<-args[3]
+# metrics keys
+metKeys<-strsplit(args[4],split=',')[[1]]
+print(metKeys)
+# checking data format and header
+met1<-read.table(metfile1,header=T,sep=',',stringsAsFactors=F)
+met2<-read.table(metfile2,header=T,sep=',',stringsAsFactors=F)
+colnames(met1)[1]<-'metrics'
+colnames(met2)[1]<-'metrics'
+
+colnames1<-colnames(met1)
+colnames2<-colnames(met2)
+mlist<-match(colnames1,colnames2)
+nalist<-is.na(mlist)
+if(sum(nalist)>0){
+  print("input files have discrepency columns")
+  exit()
+}
+# re-order 
+met2<-met2[,mlist]
+# select subset of metrics 
+mlist1<-match(metKeys,met1$metrics)
+met1.core<-met1[mlist1,]
+mlist2<-match(metKeys,met2$metrics)
+met2.core<-met2[mlist2,]
+out<-c()
+pouts<-list()
+for(ii in 1:length(metKeys)){
+  x<-as.numeric(met1.core[ii,-1])
+  y<-as.numeric(met2.core[ii,-1])
+  z<-data.frame('Base'=x,'Updated'=y)
+  # linear regression model
+  fit<-lm(y~x)
+  sfit<- summary(fit)
```

Comment:
I'm not sure what "standard R style" is (can you link me to something? I'd love to know. 

... in any case, you should move the space here to be consistent with the rest of your work in this function.